File: lab04/java/main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int opcao;

        do {
            System.out.println("\nEscolha a opção:");
            System.out.println("1 - Criar Gerente");
            System.out.println("2 - Criar Desenvolvedor");
            System.out.println("3 - Criar TechLead");
            System.out.println("4 - Sair");
            System.out.print("Opção: ");
            opcao = scanner.nextInt();
            scanner.nextLine();

            switch (opcao) {
                case 1:
                    System.out.print("Digite o nome do Gerente: ");
                    String nomeGerente = scanner.nextLine();
                    System.out.print("Digite o salário base do Gerente: ");
                    double salarioGerente = scanner.nextDouble();
                    System.out.print("Digite o número de equipes do Gerente: ");
                    int numeroDeEquipesGerente = scanner.nextInt();
                    GerenteFun gerente = new GerenteFun(nomeGerente, salarioGerente, numeroDeEquipesGerente);
                    System.out.println(gerente.descricao());
                    break;

                case 2:
                    System.out.print("Digite o nome do Desenvolvedor: ");
                    String nomeDesenvolvedor = scanner.nextLine();
                    System.out.print("Digite o salário base do Desenvolvedor: ");
                    double salarioDesenvolvedor = scanner.nextDouble();
                    scanner.nextLine();
                    System.out.print("Digite a linguagem principal do Desenvolvedor: ");
                    String linguagemDesenvolvedor = scanner.nextLine();
                    DesenvolvedorFun desenvolvedor = new DesenvolvedorFun(nomeDesenvolvedor, salarioDesenvolvedor, linguagemDesenvolvedor);
                    System.out.println(desenvolvedor.descricao());
                    break;

                case 3:
                    System.out.print("Digite o nome do TechLead: ");
                    String nomeTechLead = scanner.nextLine();
                    System.out.print("Digite o salário base do TechLead: ");
                    double salarioTechLead = scanner.nextDouble();
                    System.out.print("Digite o número de equipes do TechLead: ");
                    int numeroDeEquipesTechLead = scanner.nextInt();
                    scanner.nextLine();
                    System.out.print("Digite a linguagem principal do TechLead: ");
                    String linguagemTechLead = scanner.nextLine();
                    System.out.print("Digite o número de projetos em andamento do TechLead: ");
                    int projetosEmAndamento = scanner.nextInt();
                    TechLead techLead = new TechLead(nomeTechLead, salarioTechLead, numeroDeEquipesTechLead, linguagemTechLead, projetosEmAndamento);
                    System.out.println(techLead.descricao());
                    break;

                case 4:
                    System.out.println("Saindo...");
                    break;

                default:
                    System.out.println("Opção inválida! Tente novamente.");
            }

        } while (opcao != 4);

        scanner.close();
    }
}

File: lab04/c++/main.cpp
#include <iostream>
#include <vector> 

#include "Gerente.hpp"
#include "Desenvolvedor.hpp"
#include "TechLead.hpp"

using namespace std;

int main(){
       
    vector <Funcionario*> funcionarios;
    int op;

    do {
        cout << "*** MENU ***\n";
        cout << "1. Adicionar Gerente\n";
        cout << "2. Adicionar Desenvolvedor\n";
        cout << "3. Adicionar TechLead\n";
        cout << "4. Mostrar Descrições\n";
        cout << "5. Sair\n";
        cin >> op;  

        switch (op){
            case 1:{
                string nome;
                float salario;
                int equipes;
                
                cout << "Digite o nome do gerente: ";
                cin.ignore();
                getline(cin, nome);

                cout << "Digite o salario base: ";
                cin >> salario;

                cout << "Digite o número de equipes sob supervisão: ";
                cin >> equipes;

                funcionarios.push_back(new Gerente(nome, salario, equipes));
                cout << "Gerente adicionado com sucesso!\n";
                break;
            }
            case 2:{
                string nome, linguagem; 
                float salario;

                cout << "Digite o nome do Desenvolvedor: ";
                cin.ignore();
                getline(cin, nome);

                cout << "Digite o salário base: ";
                cin >> salario;

                cout << "Digite a linguagem principal: ";
                cin.ignore();
                getline(cin, linguagem);
                
                funcionarios.push_back(new Desenvolvedor(nome, salario, linguagem));
                cout << "Desenvolvedor adicionado com sucesso!\n";
                break;
            }

            case 3:{

                string nome, linguagem;
                double salario;
                int equipes, projetos;

                cout << "Digite o nome do TechLead: ";
                cin.ignore();
                getline(cin, nome);

                cout << "Digite o salário base: ";
                cin >> salario;

                cout << "Digite o número de equipes sob supervisão: ";
                cin >> equipes;

                cout << "Digite a linguagem principal: ";
                cin.ignore();
                getline(cin, linguagem);

                cout << "Digite o número de projetos em andamento: ";
                cin >> projetos;

                funcionarios.push_back(new TechLead(projetos, linguagem, equipes, nome, salario));
                cout << "TechLead adicionado com sucesso!\n";
                break;
            }

            case 4:{
                cout << "\n***DESCRIÇÕES***\n";
                for(size_t i = 0; i < funcionarios.size(); i++){
                    cout << funcionarios [i] ->descricao() << endl;
                }
                break;
            }

            case 5:{
                cout << "Encerrando\n";
                break;
            }

            default:{
                cout << "Opção inválida\n";
                break;
            }
        }

    }while (op != 5);
        
        for (auto& func : funcionarios){
            delete func;
        }

    return 0;

}